site/src/examples/wealth-and-health-of-nations/index.js

```diff
@@ -0,0 +1,232 @@
+/* global d3:false, fc:false, example: false */
+(function(d3, fc, example) {
+
+    // Inspired by: http://bost.ocks.org/mike/nations/
+
+    example.yearLabel = function() {
+        var event = d3.dispatch('yearChange');
+
+        var labelBox;
+
+        var yearScale = d3.scale.linear()
+              .domain([1800, 2009])
+              .clamp(true);
+
+        var overlayDataJoin = fc.util.dataJoin()
+            .selector('rect.overlay')
+            .element('rect')
+            .attr('class', 'overlay');
+
+        var labelDataJoin = fc.util.dataJoin()
+            .selector('text')
+            .element('text');
+
+        function mousemove() { event.yearChange(yearScale.invert(d3.mouse(this)[0])); }
+
+        function yearLabel(selection) {
+            selection.each(function(data) {
+                var label = labelDataJoin(selection, [data]);
+
+                label.enter()
+                    .attr('text-anchor', 'end');
+
+                label.text(data.year);
+
+                var overlay = overlayDataJoin(selection, [data]);
+
+                overlay.enter()
+                    .each(function() {
+                        labelBox = label.node().getBBox();
```

Comment:
that's a neat idea, grabbing the BBox via `each` to keep the whole thing 'fluent'